### PR TITLE
fix code smells identified by sonar

### PR DESCRIPTION
**Pull request by shubhbapna**
Fixes all code smells identifies by sonar analysis

### CODE REVIEW

**Review by rgdoliveira — APPROVED** [review]

**Review by lampajr — APPROVED** [review]
lgtm

File: src/util/mapping.ts [hunk 1]
```diff
@@ -71,14 +71,12 @@ export function getMapping(
   }
 
   // if no mapping from starting project's mapping is found, try current project's mapping
-  const mappingFromCurrentProject = getSourceToTargetFromProjectOrDefault(
+  return getSourceToTargetFromProjectOrDefault(
```
Comment (Ginxo) [inline src/util/mapping.ts:74]: wdyt?
```
   return getSourceToTargetFromProjectOrDefault(
    targetBranch,
    currentProject,
    startingProjectMappings.exclude,
    startingProjectMappings.dependencies
  ) || getSourceToTargetFromProjectOrDefault(
    targetBranch,
    startingProject,
    currentProjectMappings.exclude,
    currentProjectMappings.dependant
  );
```

File: src/util/yaml.ts [hunk 2]
```diff
@@ -22,26 +22,30 @@ export async function validateDefinitionFile(content: string) {
...
 /**
  * A reviver function to transform json keys
- * @param key 
- * @param value 
- * @returns 
+ * @param key
+ * @param value
+ * @returns
  */
 function reviver(key: unknown, value: unknown) {
```
Comment (Ginxo) [inline src/util/yaml.ts:29]: wdyt?
```
function reviver(key: unknown, value: unknown) {
  return convertToArray(key, value) ||
            parseMapping(key, value) ||
            parseBuild(key, value) ||
            parseArchiveArtifacts(key, value) ||
            value
}
```

**Review by Ginxo — COMMENTED** [review]
wdyt?

File: src/util/yaml.ts (hunk 2 above)
Comment (lampajr) [reply src/util/yaml.ts:29]: It is definitely more compact, but is it still equally clear (just to raise the doubt)? I would say yes in this case, therefore I agree

**Pull request merged**